File: app/src/main/java/net/daverix/ajvm/jvm/IfEqOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class IfEqOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader, int indexOfBytecode, Frame currentFrame) throws IOException {
        int ifEqOffset = reader.readUnsignedShort();
        int value = (int) currentFrame.pop();
        if (value == 0) {
            reader.jumpTo(indexOfBytecode + ifEqOffset);
        }
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/IfEqOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class IfEqOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader, indexOfBytecode: Int, currentFrame: Frame) {
        val ifEqOffset = reader.readUnsignedShort()
        val value = currentFrame.pop() as Int
        if (value == 0) {
            reader.jumpTo(indexOfBytecode + ifEqOffset)
        }
    }
}
